libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx

```diff
@@ -62,24 +64,90 @@ export default function AddEditClient({ id }: { id?: string }) {
       },
     ],
   });
-  const [updateClient] = useUpdateClientProfileMutation();
-  const [createClient] = useCreateClientProfileMutation({
-    refetchQueries: [
-      {
-        query: ClientProfilesDocument,
-        variables: {
-          pagination: { limit: 20 + 1, offset: 0 },
-          filters: {
-            search: '',
-          },
-          order: {
-            user_FirstName: Ordering.AscNullsFirst,
-            id: Ordering.Desc,
+  const [updateClient, { error: updateMutationError }] =
+    useUpdateClientProfileMutation();
+
+  const [createClient, { error: createMutationError }] =
+    useCreateClientProfileMutation({
+      refetchQueries: [
+        {
+          query: ClientProfilesDocument,
+          variables: {
+            pagination: { limit: 20 + 1, offset: 0 },
+            filters: {
+              search: '',
+            },
+            order: {
+              user_FirstName: Ordering.AscNullsFirst,
+              id: Ordering.Desc,
+            },
           },
         },
-      },
-    ],
-  });
+      ],
+    });
+
+  const extractErrorFields = (errors: ApolloError) => {
+    return errors.graphQLErrors
+      .map((error: GraphQLFormattedError) => {
+        const regex = /data\.(.*?);/;
+        const match = error.message.match(regex);
+        return match ? match[1] : null;
+      })
+      .filter(Boolean);
+  };
+
+  const getMutationError = useCallback(
+    (
+      updateError: typeof updateMutationError,
+      createError: typeof createMutationError
+    ) => {
+      if (updateError) return updateError;
+      if (createError) return createError;
+      return null;
+    },
+    []
+  );
+
+  const handleMutationError = useCallback(
+    (mutationError: ApolloError) => {
+      const extractedFields = extractErrorFields(mutationError).filter(
+        (dataField) => dataField !== null
+      );
+      const extensions = mutationError.cause?.extensions as {
+        message?: string;
+      };
+      if (extensions?.message) {
+        extractedFields.map((dataField) => {
+          return methods.setError(
+            dataField as
+              | keyof UpdateClientProfileInput
+              | keyof CreateClientProfileInput,
+            {
+              type: 'custom',
+              message: extensions.message,
+            }
```

Comment:
This forcefully sets the error within react-hook-form to the nodes referred from lines [89 - 97](https://github.com/BetterAngelsLA/monorepo/pull/549/files#r1783655683)